bottlecap/src/logs/aggregator.rs

```diff
@@ -1,11 +1,10 @@
-use serde::Serialize;
-use std::collections::VecDeque;
+use crossbeam::queue::SegQueue;
```

Comment:
FWIW SegQueue is an unbounded multi-producer, multi-consumer. Minor issue, the multi-consumer bit requires more coordination in the implementation so you're leaving performance on the table by not using a single-consumer. More pressing, this is an unbounded queue and represents a potential site for unbounded allocations.

Reply:
I'm aware of the tradeoffs, I have to clean this PR still, will mark it as draft

Reply:
SegQueue is also theoretically lock free but it uses Acquire/Release semantics internally and so the semantics are equivalent to a mutex. Except Acq/Rel done on atomic structures cuts the Linux scheduler out so your latency will be worse generally speaking, except in specialized circumstances where the acq/rel using software has the highest CPU priority anyway and is always scheduled first. 

Tokio's mpsc is good, will play well with the Linux scheduler and also nicely between async/sync code.